From 546ec73fcb143e7ddc87385e9e8327c832d862ce Mon Sep 17 00:00:00 2001
From: Colin Watson <cjwatson@ubuntu.com>
Date: Fri, 28 Mar 2014 17:09:49 +0000
Subject: Ignore zero-length devices

There seems no point in complaining about these.  The problem with
throwing an exception is that it's difficult for partitioners such as
partman to programmatically distinguish between this kind of thing and
some more serious error that actually merits letting the user know.

Bug-Debian: http://bugs.debian.org/602533
Forwarded: http://lists.gnu.org/archive/html/bug-parted/2010-11/msg00011.html
Reviewed-by: Otavio Salvador <otavio@ossystems.com.br>
Last-Update: 2010-11-19

Patch-Name: zero-length-devices.patch
---
 libparted/arch/linux.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/libparted/arch/linux.c b/libparted/arch/linux.c
index 38f4e31..3afe976 100644
--- a/libparted/arch/linux.c
+++ b/libparted/arch/linux.c
@@ -1180,6 +1180,12 @@ init_generic (PedDevice* dev, const char* model_name)
         if (_device_probe_geometry (dev)) {
                 ped_exception_leave_all ();
         } else {
+                if (!_device_get_length (dev)) {
+                        ped_exception_catch ();
+                        ped_exception_leave_all ();
+                        goto error_close_dev;
+                }
+
                 /* hack to allow use of files, for testing */
                 ped_exception_catch ();
                 ped_exception_leave_all ();
